File: packages/capp-sdk-ts/src/modules/approvals.ts
import { CAPPClient } from '../client';
import { ApprovalRequest } from '../models';
import { handleApiError } from '../utils';

export class ApprovalsModule {
    constructor(private client: CAPPClient) { }

    public async list(status: string = 'pending'): Promise<ApprovalRequest[]> {
        const res = await this.client.fetch(`/approvals?status=${encodeURIComponent(status)}`);
        await handleApiError(res);
        const data = await res.json();
        return data.items || [];
    }

    public async resolve(
        approvalId: string,
        decision: string,
        note?: string
    ): Promise<void> {
        const body: Record<string, string> = { decision };
        if (note) body.note = note;

        const res = await this.client.fetch(`/approvals/${approvalId}/resolve`, {
            method: 'POST',
            body: JSON.stringify(body)
        });
        await handleApiError(res);
    }
}

File: packages/capp-sdk/capp/modules/approvals.py
import httpx
from typing import List, Optional
from ..models import ApprovalRequest
from .._utils import handle_api_error

class ApprovalsModule:
    def __init__(self, client: httpx.AsyncClient):
        self._client = client
        
    async def list(self, status: str = "pending") -> List[ApprovalRequest]:
        res = await self._client.get("/approvals", params={"status": status})
        handle_api_error(res)
        return [ApprovalRequest(**item) for item in res.json().get("items", [])]

    async def resolve(self, approval_id: str, decision: str, note: Optional[str] = None):
        payload = {"decision": decision}
        if note:
            payload["note"] = note
            
        res = await self._client.post(f"/approvals/{approval_id}/resolve", json=payload)
        handle_api_error(res)
